### PR TITLE
🎉(global) bootstrap project

### PR DESCRIPTION
**Pull request by jbpenrath**
Configure project, a playground to develop and QA tools

### CODE REVIEW

**Comment by jbpenrath** [comment]
@sampaccoud DraftForm is now Verna.js 🎉 
image

File: .github/ISSUE_TEMPLATE/Bug_report.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+name: 🐛 Bug Report
+about: If something is not working as expected 🤔.
+
+---
+
+## Bug Report
+
+### Problematic behavior
```
Comment (sampaccoud) [inline .github/ISSUE_TEMPLATE/Bug_report.md:9]: :medal_sports: :stuck_out_tongue:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Contributing to Draft Form
+
+Want to contribute to Draft Form? We got you covered.
```
Comment (sampaccoud) [inline CONTRIBUTING.md:3]: vernajs!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Contributing to Draft Form
```
Comment (sampaccoud) [inline CONTRIBUTING.md:1]: Vernajs!

File: README.md [hunk 4]
```diff
@@ -1,3 +1,41 @@
...
+# Verna.js 🏗
+[![CircleCI](https://circleci.com/gh/openfun/vernajs/tree/master.svg?style=svg)](https://circleci.com/gh/openfun/vernajs/tree/master)
 
 An extensible form builder based on [React JSON Schema Form](https://github.com/rjsf-team/react-jsonschema-form).
+
+## Overview
+
+Verna.js is a React component using [JSON Schema](http://json-schema.org/) to build and
+generate forms. As it relies on [React JSON Schema Form](https://github.com/rjsf-team/react-jsonschema-form)
+library, it is fully extensible to fit your needs.
```
Comment (sampaccoud) [inline README.md:10]: I would say it uses json schema but generates json schema as output...
```suggestion
Verna.js is a React component to build forms that generates a [JSON Schema](http://json-schema.org/) description of the form as output. The form can then be rendered with the [React JSON Schema Form](https://github.com/rjsf-team/react-jsonschema-form)
library, which makes it fully extensible to fit your needs.
```

File: README.md [hunk 5]
```diff
@@ -1,3 +1,41 @@
...
+
+
+## Quick preview
+
+You can check examples available within `examples` folder. Take a loot at `README.md` in
```
Comment (sampaccoud) [inline README.md:15]:
```suggestion
You can check examples available within the `examples` folder. Take a loot at the `README.md` file in
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,41 @@
...
+### Instructions
+
+🚧 **`@openfun/vernajs` is not yet available on NPM. Stay tuned !**
+Soon instructions to install `@openfun/vernajs` within your project.
+
+To run the project, you have to run :
+1. `npm install` or `yarn`
+2. `npm start` or `yarn start`
+
+You should now be able to access to the playground example from http://localhost:3000
```
Comment (sampaccoud) [inline README.md:27]:
```suggestion
You should now be able to access the playground example from http://localhost:3000
```

File: examples/playground/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Draft Form - Examples
```
Comment (sampaccoud) [inline examples/playground/README.md:1]: vernajs!

File: examples/playground/index.html [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>🏗 VernaForm Playground</title>
```
Comment (sampaccoud) [inline examples/playground/index.html:6]:
```suggestion
    <title>🏗 Verna.js Forms Playground</title>
```

File: examples/playground/src/App.tsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import DrafFrom from '@openfun/vernajs';
+
+function App() {
+  return <DrafFrom />;
```
Comment (sampaccoud) [inline examples/playground/src/App.tsx:4]: Should it be something like:
```suggestion
  return <VernaForm />;
```

File: jest.setup.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+// Extend jest matchers with jest-dom's
+import '@testing-library/jest-dom/extend-expect';
```
Comment (sampaccoud) [inline jest.setup.js:2]: missing blank line

File: src/VernaForm.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    screen.getByRole('group', { name: 'A registration form' });
+
+    // - A required text input First name should be displayed with a default value Richie
+    const $firstnameField = screen.getByRole('textbox', {
+      name: 'First name *',
+    }) as HTMLInputElement;
+    expect($firstnameField.required).toEqual(true);
+    expect($firstnameField.value).toEqual('Richie');
+
+    // - A required text input First name should be displayed with a default value Richie
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:18]:
```suggestion
    // - A required text input Last name should be displayed with no default value
```

File: src/VernaForm.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    const $lastnameField = screen.getByRole('textbox', { name: 'Last name *' }) as HTMLInputElement;
+    expect($lastnameField.required).toEqual(true);
+
+    const $phoneField = screen.getByRole('textbox', { name: 'Telephone' }) as HTMLInputElement;
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:22]: add a comment also?

File: src/VernaForm.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
+import { screen, render } from '@testing-library/react';
+import VernaForm from '.';
+
+describe('VernaForm', () => {
+  it('should render', async () => {
+    render(<VernaForm />);
+
+    // - A required text input First name should be displayed with a default value Richie
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:8]:
```suggestion
copy/pasta fail?
```

File: src/VernaForm.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    screen.getByRole('group', { name: 'A registration form' });
+
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:10]: test also the description?

File: src/VernaForm.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    // - A required text input First name should be displayed with a default value Richie
+    const $firstnameField = screen.getByRole('textbox', {
+      name: 'First name *',
+    }) as HTMLInputElement;
+    expect($firstnameField.required).toEqual(true);
+    expect($firstnameField.value).toEqual('Richie');
+
+    // - A required text input First name should be displayed with a default value Richie
+    const $lastnameField = screen.getByRole('textbox', { name: 'Last name *' }) as HTMLInputElement;
+    expect($lastnameField.required).toEqual(true);
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:20]: Add 
```javascript
expect($firstnameField.value).toEqual('');
```

File: src/VernaForm.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+    const $phoneField = screen.getByRole('textbox', { name: 'Telephone' }) as HTMLInputElement;
+    expect($phoneField.required).toEqual(false);
```
Comment (sampaccoud) [inline src/VernaForm.spec.tsx:23]: Add:
```
expect($phoneField.value).toEqual('');
```

File: src/VernaForm.tsx [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    firstName: {
+      type: 'string',
+      title: 'First name',
+      default: 'Richie',
+    },
+    lastName: {
+      type: 'string',
+      title: 'Last name',
+    },
+    telephone: {
```
Comment (sampaccoud) [inline src/VernaForm.tsx:19]:
```suggestion
    phone: {
```

File: src/VernaForm.spec.tsx (hunk 16 above)
Comment (jbpenrath) [reply src/VernaForm.spec.tsx:23]: Nope a jQuery nostalgic when I manipulate DOM elements 😅

File: examples/playground/index.html (hunk 8 above)
Comment (jbpenrath) [reply examples/playground/index.html:6]: I just replaced by `Verna.js Playground`

File: examples/playground/src/App.tsx (hunk 9 above)
Comment (jbpenrath) [reply examples/playground/src/App.tsx:4]: Yep the search and replace did not catch it.

File: src/VernaForm.spec.tsx (hunk 16 above)
Comment (sampaccoud) [reply src/VernaForm.spec.tsx:28]: I forgot that :astonished:

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**